src/ImageRunModal.jsx

```diff
@@ -934,15 +970,18 @@
                                  formclass='publish-port-form'
                                  label={_("Port mapping")}
                                  actionLabel={_("Add port mapping")}
+                                 validationFailed={dialogValues.validationFailed.publish}
+                                 onValidationChange={value => this.dynamicListOnValidationChange(value, "publish")}
                                  onChange={value => this.onValueChanged('publish', value)}
                                  default={{ IP: null, containerPort: null, hostPort: null, protocol: 'tcp' }}
                                  itemcomponent={ <PublishPort />} />
-
                         <DynamicListForm id='run-image-dialog-volume'
                                  emptyStateString={_("No volumes specified")}
                                  formclass='volume-form'
                                  label={_("Volumes")}
                                  actionLabel={_("Add volume")}
+                                 validationFailed={dialogValues.validationFailed.volumes}
+                                 onValidationChange={value => this.dynamicListOnValidationChange(value, "volumes")}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20230907-093411-a9f37bd4-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)